DQMOffline/Lumi/plugins/ZCounting.cc

```diff
@@ -389,7 +438,225 @@ void ZCounting::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
   }//End of tag loop
 
 }
+void ZCounting::analyzeElectrons(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  edm::LogInfo("ZCounting") <<  "ZCounting::analyzeElectrons" << std::endl;
+
+  //-------------------------------
+  //--- Vertex
+  //-------------------------------
+  edm::Handle<reco::VertexCollection> hVertexProduct;
+  iEvent.getByToken(fPVName_token,hVertexProduct);
+  if(!hVertexProduct.isValid()) return;
+
+  const reco::VertexCollection *pvCol = hVertexProduct.product();
+  int nvtx = 0;
+
+  for(reco::VertexCollection::const_iterator itVtx = pvCol->begin(); itVtx!=pvCol->end(); ++itVtx) {
+    if(itVtx->isFake())                             continue;
+    if(itVtx->tracksSize()     < VtxNTracksFitCut_) continue;
+    if(itVtx->ndof()           < VtxNdofCut_)       continue;
+    if(fabs(itVtx->z())        > VtxAbsZCut_)       continue;
+    if(itVtx->position().Rho() > VtxRhoCut_)        continue;
+
+    nvtx++;
+  }
+
+  // Good vertex requirement
+  if(nvtx==0) return;
 
+  //-------------------------------
+  //--- Trigger
+  //-------------------------------
+  edm::Handle<edm::TriggerResults> hTrgRes;
+  iEvent.getByToken(fHLTTag_token,hTrgRes);
+  if(!hTrgRes.isValid()) return;
+
+  edm::Handle<trigger::TriggerEvent> hTrgEvt;
+  iEvent.getByToken(fHLTObjTag_token,hTrgEvt);
+
+
+  const edm::TriggerNames &triggerNames = iEvent.triggerNames(*hTrgRes);
+  Bool_t config_changed = false;
+  if(fTriggerNamesID != triggerNames.parameterSetID()) {
+    fTriggerNamesID = triggerNames.parameterSetID();
+    config_changed  = true;
+  }
+  if(config_changed) {
+    initHLT(*hTrgRes, triggerNames);
+  }
+
+  TriggerBits triggerBits;
+  for(unsigned int irec=0; irec<fTrigger->fRecords.size(); irec++) {
+    if(fTrigger->fRecords[irec].hltPathIndex == (unsigned int)-1) continue;
+    if(hTrgRes->accept(fTrigger->fRecords[irec].hltPathIndex)) {
+      triggerBits [fTrigger->fRecords[irec].baconTrigBit] = true;
+    }
+  }
+
+  // Trigger requirement
+  if(!isElectronTrigger(*fTrigger, triggerBits)) return;
+
+  // Get Electrons
+  edm::Handle<edm::View<reco::GsfElectron> > electrons;
+  iEvent.getByToken(fGsfElectronName_token, electrons);
+
+  // Get SuperClusters
+  edm::Handle<edm::View<reco::SuperCluster> > superclusters;
+  iEvent.getByToken(fSCName_token, superclusters);
+
+  // Get Rho
+  edm::Handle<double> rhoHandle;
+  iEvent.getByToken(fRhoToken, rhoHandle);
+  EleID_.setRho(*rhoHandle);
+
+  // Get beamspot
+  edm::Handle<reco::BeamSpot> beamspotHandle;
+  iEvent.getByToken(fBeamspotToken, beamspotHandle);
+  EleID_.setBeamspot(beamspotHandle);
+
+  // Conversions
+  edm::Handle<reco::ConversionCollection> conversionsHandle;
+  iEvent.getByToken(fConversionToken, conversionsHandle);
+  EleID_.setConversions(conversionsHandle);
+
+  TLorentzVector vTag(0.,0.,0.,0.);
+  TLorentzVector vProbe(0.,0.,0.,0.);
+  TLorentzVector vDilep(0.,0.,0.,0.);
```

Comment:
move the declaration of these TLorentzVectors to where they are used. Its better to avoid these copies completely - I'm not sure its possible yet. I'll follow up.

Reply:
Done.